autoscheduler/frontend/src/components/SchedulingPage/SchedulePreview/MiniSchedule/MiniSchedule.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import Meeting from '../../../../types/Meeting';
+import { LAST_HOUR, FIRST_HOUR } from '../../../../timeUtil';
+import DayOfWeek from '../../../../types/DayOfWeek';
+import * as parentStyles from '../../Schedule/Schedule.css';
+import * as styles from './MiniSchedule.css';
+import colors from '../../Schedule/meetingColors';
+
+interface MiniScheduleProps {
+  schedule: Meeting[];
+}
+
+const MiniSchedule: React.FC<MiniScheduleProps> = ({ schedule }) => {
+  // these must be unique because of how they're used below
+  const DAYS_OF_WEEK = ['M', 'T', 'W', 'R', 'F'];
+
+  // build rows from first and last hour
+  const HOURS_OF_DAY = [];
+  for (let h = FIRST_HOUR; h <= LAST_HOUR; h++) { HOURS_OF_DAY.push(h); }
+  const hourBars = HOURS_OF_DAY.map((hour) => (
+    <div className={parentStyles.calendarRow} key={hour}>
+      <div className={styles.hourMarker} style={{ top: 0 }} />
+    </div>
+  ));
+
+  // let's see if useMemo reduces our render time
+  const meetingsForDays = React.useMemo(() => {
+    const uniqueSections = [...new Set(schedule.map((mtg: Meeting) => mtg.section.id))];
+    // build each day based on schedule
+    function getMeetingsForDay(day: number): Meeting[] {
+      // meetingDays = UMTWRFS
+      // day = MTWRF
+      return schedule.filter((meeting) => meeting.meetingDays[day]);
+    }
+    function renderMeeting(meeting: Meeting): JSX.Element {
+      const {
+        endTimeHours, endTimeMinutes, startTimeHours, startTimeMinutes,
+      } = meeting;
+      const elapsedTime = endTimeHours * 60 + endTimeMinutes
+        - startTimeHours * 60 - startTimeMinutes;
+      const computedStyle: React.CSSProperties = {
+        position: 'absolute',
+        width: '100%',
```

Comment:
Since we have to calculate three of the CSS attributes dynamically anyway, I think it's better to have all of the styling in one place. That way, we don't accidentally end up with conflicting styles in `.css` and `.tsx` files